### PR TITLE
Feature: Manage wallets modal component

### PR DESCRIPTION
**Pull request by RakeshUP**
## Description

Task: [APP-561](https://aragonassociation.atlassian.net/jira/software/c/projects/APP/boards/40?modal=detail&selectedIssue=APP-561)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: packages/web-app/src/containers/manageWalletsModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        </SelectAllContainer>
+
+        <div className="space-y-1.5">
+          {filteredWallets.map(wallet => (
+            <CheckboxListItem
+              key={wallet}
+              label={shortenAddress(wallet)}
+              multiSelect
+              state={selectedWallets[wallet] ? 'active' : 'default'}
+              onClick={() => {
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/manageWalletsModal/index.tsx]: If you've the time, can you maybe move this up top into a handler?

File: packages/web-app/src/containers/manageWalletsModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      return wallets.reduce((tempSelectedWallets, wallet) => {
+        wallet.match(re) && tempSelectedWallets.push(wallet);
+        return tempSelectedWallets;
+      }, [] as Array<string>);
+    } else {
+      return wallets;
+    }
+  }, [searchValue]);
+
+  const handleSelectAll = () => {
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/manageWalletsModal/index.tsx:63]: I wonder if there's a way to unselect all without well, unselecting all one by one 🤔

File: packages/web-app/src/containers/manageWalletsModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        <SearchInput
+          value={searchValue}
+          placeholder={t('placeHolders.searchTokens')}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+            setSearchValue(e.target.value)
+          }
+        />
+      </ModalHeader>
+      <Container>
+        <SelectAllContainer>
```
Comment (Fabricevladimir) [inline packages/web-app/src/containers/manageWalletsModal/index.tsx:96]: There is a `no address selected` text that seems to be present on the designs. I'm assuming it's to be shown when no addresses are selected.

**Review by Fabricevladimir — APPROVED** [review]
Thanks. Left a couple of comments

File: packages/web-app/src/containers/manageWalletsModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        </div>
+      </Container>
+
+      <ButtonContainer>
+        <ButtonText
+          label={`Add ${Object.keys(selectedWallets).length} Wallets`}
+          size="large"
+          onClick={() => {
+            addWalletCallback(Object.keys(selectedWallets));
+            close('manageWallet');
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/manageWalletsModal/index.tsx]: FYI: The datastructure that holds the selected addresses is not reset when the modal is closed. Thus, the addresses remain selected when reopening the modal. I'm not saying this is nevery useful, but I'm pretty sure there's situations where this is not what should happen. Maybe add a flag that allows the caller to decide what happens on close?

**Review by WissenIstNacht — COMMENTED** [review]
Thanks for the PR!

If you disagree with the comment please let me know and I'll approve the PR.

File: packages/web-app/src/containers/manageWalletsModal/index.tsx (hunk 2 above)
Comment (RakeshUP) [reply packages/web-app/src/containers/manageWalletsModal/index.tsx:63]: IMO, the "select all" should've been a button and after clicking it, changes to "unselect all". I'll talk to Selim/Marco about this

File: packages/web-app/src/containers/manageWalletsModal/index.tsx (hunk 4 above)
Comment (RakeshUP) [reply packages/web-app/src/containers/manageWalletsModal/index.tsx]: Added a resetOnClose flag!

**Review by WissenIstNacht — APPROVED** [review]
Thanks for the changes!

**Pull request merged**